c/cmake_modules/BuildUtils.cmake

```diff
@@ -240,6 +240,21 @@ function(ADD_ARROW_LIB LIB_NAME)
                                      OUTPUT_NAME ${LIB_NAME}
                                      VERSION "${ADBC_FULL_SO_VERSION}"
                                      SOVERSION "${ADBC_SO_VERSION}")
+                                     
+    if(MSVC)
```

Comment:
Should we use `WIN32` not `MSVC`?

Reply:
I was torn between the two, leaned more towards MSVC as the compiler choice is what's driving the behaviour. Happy to use WIN32 though, both will lead to the same result.

Reply:
It seems that we need this for mingw-w64. So I feel `WIN32` is better.

Reply:
No worries, changed to WIN32. Same goes to the rest.